### PR TITLE
contracts for evm

### PR DESCRIPTION
**Pull request by Tri-stone**
## Description

What is the purpose of the change?

Support contracts for evm.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Brief of your solution

Support contracts for evm.

1. Deploy, invoke and query an EVM contract;
2. Transfer native token for Xuper;
3. Cross contract call function between solidity contracts;

## How Has This Been Tested?

Please test these functions with relative contracts.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -38,15 +40,17 @@ require (
...
 	golang.org/x/sync v0.0.0-20190423024810-112230192c58
 	google.golang.org/genproto v0.0.0-20190927181202-20e1ac93f88c
-	google.golang.org/grpc v1.24.0
+	google.golang.org/grpc v1.27.1
 	gopkg.in/sourcemap.v1 v1.0.5 // indirect
-	gopkg.in/yaml.v2 v2.2.2
+	gopkg.in/yaml.v2 v2.2.4
 )
+
+replace github.com/hyperledger/burrow => github.com/Tri-stone/burrow v0.30.18
```
Comment (qizheng09) [inline go.mod:56]: Burrow has been merged, plz change to the repo.

File: core/contract/evm/creator.go [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+		Gas:      &gas,
+	}
+	out, err := e.vm.Execute(e.state, e.blockState, e, params, e.code)
+	if err != nil {
+		return err
+	}
+
+	e.gasUsed = uint64(contract.MaxLimits.Cpu) - *params.Gas
+
+	if e.ctx.Method != "" {
```
Comment (icexin) [inline core/contract/evm/creator.go:133]: It is recommended not to do abi decode here, leave it to cli or sdk to do

File: core/contract/evm/state.go [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+		Permissions: permission.AllAccountPermissions,
+	}, nil
+}
+
+// Retrieve a 32-byte value stored at key for the account at address, return Zero256 if key does not exist but
+// error if address does not
+func (s *stateManager) GetStorage(address crypto.Address, key binary.Word256) ([]byte, error) {
+	fmt.Printf("\nget %s %s\n", address, key)
+	fmt.Printf("first address %s\n", s.ctx.ContractName)
+	contractName, _, err := DetermineEVMAddress(address)
```
Comment (icexin) [inline core/contract/evm/state.go:59]: This function will be called frequently, can the address here be asserted as a contract address

**Review by icexin — APPROVED** [review]
LGTM

**Review by chunhui01 — APPROVED** [review]
LGTM

**Review by qizheng09 — APPROVED** [review]
LGTM

**Pull request merged**